core/src/graph/graph.rs

```diff
@@ -411,13 +413,34 @@ impl Graph {
 		Ok(None)
 	}
 
+	/// returns one sided friendship connections
+	pub fn get_one_sided_friendships(&self) -> Result<Vec<DsnpGraphEdge>> {
+		if self.get_connection_type() != ConnectionType::Friendship(PrivacyType::Private) {
+			return Err(Error::msg(
+				"Calling get_all_one_sided_friendships in non private friendship graph!",
+			))
+		}
```

Comment:
Technically, a public friendship could also be one-sided, it just doesn't require PRId.

Reply:
I don't think we have any public friendship use-case currently but will create a ticket to track it in case it comes up in future.